server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -889,7 +889,7 @@ private Specification<Candidate> computeQuery(
         return query;
     }
 
-    private BoolQueryBuilder computeElasticQuery(BoolQueryBuilder boolQueryBuilder,
+    private BoolQueryBuilder computeElasticQuery(
```

Comment:
Why did we remove the builder from the call?

Reply:
Because in my previous go I was adding the keyword and filters to my original query one-by-one. That's what caused the issue — the keyword clause was being added as a MUST (which is what we want) but the filters were being combined into a single query with the original (not what we want at all and I can't believe I missed it). Now I'm building a separate query and adding it altogether to the original query as an additional MUST clause.

So, the bit you've commented on, I build the first query. Then for any joins, I compute them as a MUST addition to the first. I'll tag you on that bit.

Reply:
OK - sounds good. Thanks for the explanation.